### PR TITLE
Backport of url_sig plugin from master to 7.1.x

### PR DESCRIPTION
**Pull request by JBevillC**
Back ports #2914, "Add path param url signing option and client ipv6 verification to url_sig plugin" to 7.1.x.  The other cherry picks are previous updates to url_sig on master and are required for #2914 to resolve merge conflicts to 7.1.x and to fix autests  for #2914 on the 7.1.x branch.

- Path Param url signing and ipv6 verification from @jrushford (5cbdf5e)
- Add pparam to url_sig plugin to auth pristine url, remove scheme check, etc. from  @ywkaras (eaec3ff)
- Fix logic for loading config files from @ywkaras (57b411f)
- Coverity CID \#1367529 - Copy into fixed sized buffer (66a5af7)
- Remove use of netstat in tests (87a873c)

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci]

**Comment by jrushford** [comment]
[approve ci autest]

**Comment by jrushford** [comment]
@zwoop and @dragon512 autest is failing because 'netstat isn't available to tests/gold_tests/logging/ccid_ctid.test.py

**Comment by dragon512** [comment]
@jrushford  Can you update your version of ccid_ctid.test.py to match what is on Master branch? I think that should fix you issue. netstat should be removed 100% at this point and we should be using the building support in AuTest instead

**Comment by zwoop** [comment]
Does this include #2914 as well ? Or do we need that one and this one?

**Comment by JBevillC** [comment]
Thanks for the suggestions everyone.  I pushed a cherry-pick from @dragon512 (87a873c) that updates the ccid_ctid.test.py file to what is in master and also some small changes to the url_sig.test.py file (also the same as master).

**Comment by JBevillC** [comment]
@zwoop #2914 is already included (40039a3).  Thanks for your help

**Comment by jrushford** [comment]
This includes #2914 for backport to 7.1

**Comment by jrushford** [comment]
Thanks @dragon512, that fixed the autest failure.

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by jrushford** [comment]
[approve ci autest]

**Comment by dragon512** [comment]
[approve ci autest]

**Comment by JBevillC** [comment]
I pushed a cherry-pick (5f32608) from @ywkaras that fixes the failing ccid_ctid autest.

**Comment by JBevillC** [comment]
[ci autest]

**Comment by zwoop** [comment]
[approve ci]

File: tests/gold_tests/logging/ccid_ctid.test.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+log.ascii {
+  Format = custom,
+  Filename = 'test_ccid_ctid'
+}'''.split("\n")
+)
+
+# Ask the OS if the port is ready for connect()
+#
+def CheckPort(Port):
```
Comment (dragon512) [inline tests/gold_tests/logging/ccid_ctid.test.py]: Remove this

File: tests/gold_tests/logging/ccid_ctid.test.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    return lambda: 0 == subprocess.call('netstat --listen --tcp -n | grep -q :{}'.format(Port), shell=True)
+
+tr = Test.AddTestRun()
+# Delay on readiness of ssl port
+tr.Processes.Default.StartBefore(Test.Processes.ts, ready=CheckPort(ts.Variables.ssl_port))
```
Comment (dragon512) [inline tests/gold_tests/logging/ccid_ctid.test.py]: Change to 

tr.Processes.Default.StartBefore(Test.Processes.ts, ready=When.PortOpen(ts.Variables.ssl_port))

**Comment by dragon512** [comment]
I think if you change what I suggest this will work better for you

**Comment by dragon512** [comment]
@JBevillC Welcome! 
Let me know if you still have issues...

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
Wooohoo, more build issues ... :-)

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
@JBevillC Where are we with this? I would like to make a 7.1.3 RC soon, so if this should go in, better get cracking! ;-).

**Comment by zwoop** [comment]
[approve ci]

**Comment by JBevillC** [comment]
@zwoop I just pushed a cherry-pick that fixes the clang build on ubuntu and freebsd. All autest pass for me on ubuntu and fedora:27

**Comment by zwoop** [comment]
[approve ci].

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci autest]

**Pull request merged**